### PR TITLE
CalorimeterIslandCluster: add transverseEnergyProfile{Metric,Scale} parameters to govern the cluster energy splitting

### PR DESCRIPTION
**Pull request by veprbl**
### Briefly, what does this PR introduce?

Allows fine tuning cluster splitting. More importantly, this prevents cluster splitting from crashing when used together with adjacencyMatrix-based configuration. The issue with the current splitting is that it tries to use the same metric as used to determine neighbors in distance-based adjacency configuration, which leads to a call via uninitialized `CalorimeterIslandCluster::hitsDist` function pointer.

### What kind of change does this PR introduce?
- [x] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No

### Does this PR change default behavior?
No

### CODE REVIEW

**Comment by dhevang** [comment]
> 
I suppose you'll need to do some modifications to the LUMISPECCAL folder like you did for the other detectors?

**Comment by veprbl** [comment]
It looks that this was broken even at the time #622 was merged. I've opened #652 to track that.

**Review by dhevang — COMMENTED** [review]
The changes look fine to me but I'm not an expert in the details.
The main thing for me is that now I see the expected cluster splitting in the Lumi spectrometer CAL: eicrecon-gcc-gun-EcalLumiSpec.
I'll leave it to the other reviewers to give final approval.

**Review by Chao1009 — APPROVED** [review]
This looks good to me.

File: src/detectors/EEMC/ProtoCluster_factory_EcalEndcapNIslandProtoClusters.h [hunk 1]
```diff
@@ -49,11 +45,15 @@ class ProtoCluster_factory_EcalEndcapNIslandProtoClusters : public eicrecon::JFa
         u_globalDistEtaPhi={};//{this, "globalDistEtaPhi", {}};
         u_dimScaledLocalDistXY={};
 
+        m_splitCluster = true;
+        m_minClusterHitEdep=1.0 * dd4hep::MeV;
+        m_minClusterCenterEdep=30.0 * dd4hep::MeV;
+        u_transverseEnergyProfileMetric = "globalDistEtaPhi";
+        u_transverseEnergyProfileScale = 0.08;
```
[the remark this answers is not in the document]
Comment (veprbl) [reply src/detectors/EEMC/ProtoCluster_factory_EcalEndcapNIslandProtoClusters.h:52]: This value was obtained by running a version of https://github.com/eic/EICrecon/pull/561 on a sample of simulated pi0's and collecting "optimal" values of lambda. Unfortunately, there is no precise science behind determining this value, yet. Good news is that the value itself seems to be quite forgiving with respect to what can be used.

[the remark this answers is not in the document]
Comment (johnny8266) [reply src/detectors/EEMC/ProtoCluster_factory_EcalEndcapNIslandProtoClusters.h:52]: Got it. just curious how it comes from. Thanks!

**Pull request merged**